File: on_device_ai/lib/models/chat_models.dart
class ChatMessage {
  final String id;
  final String text;
  final bool isUser;
  final DateTime timestamp;
  final bool isLoading;

  ChatMessage({
    String? id,
    required this.text,
    required this.isUser,
    DateTime? timestamp,
    this.isLoading = false,
  })  : id = id ?? DateTime.now().millisecondsSinceEpoch.toString(),
        timestamp = timestamp ?? DateTime.now();

  ChatMessage copyWith({
    String? id,
    String? text,
    bool? isUser,
    DateTime? timestamp,
    bool? isLoading,
  }) {
    return ChatMessage(
      id: id ?? this.id,
      text: text ?? this.text,
      isUser: isUser ?? this.isUser,
      timestamp: timestamp ?? this.timestamp,
      isLoading: isLoading ?? this.isLoading,
    );
  }
}

class ModelConfig {
  final String modelPath;
  final String modelName;
  final int contextSize;
  final int threads;
  final int gpuLayers;

  const ModelConfig({
    required this.modelPath,
    this.modelName = 'Mistral 7B Instruct Q4_K_M',
    this.contextSize = 4096,
    this.threads = 4,
    this.gpuLayers = 0,
  });
}

class LLMState {
  final bool isLoading;
  final bool isModelLoaded;
  final String? error;
  final List<ChatMessage> messages;

  const LLMState({
    this.isLoading = false,
    this.isModelLoaded = false,
    this.error,
    this.messages = const [],
  });

  LLMState copyWith({
    bool? isLoading,
    bool? isModelLoaded,
    String? error,
    List<ChatMessage>? messages,
  }) {
    return LLMState(
      isLoading: isLoading ?? this.isLoading,
      isModelLoaded: isModelLoaded ?? this.isModelLoaded,
      error: error,
      messages: messages ?? this.messages,
    );
  }
}

File: on_device_ai/android/app/src/main/kotlin/com/example/on_device_ai/models/ChatModels.kt
package com.example.on_device_ai.models

import kotlinx.serialization.Serializable

@Serializable
data class ChatMessage(
    val id: String = java.util.UUID.randomUUID().toString(),
    val text: String,
    val isUser: Boolean,
    val timestamp: Long = System.currentTimeMillis(),
    val isLoading: Boolean = false
)

@Serializable
data class ModelConfig(
    val modelPath: String,
    val modelName: String = "Mistral 7B Q4_K_M",
    val contextSize: Int = 4096,
    val threads: Int = 4,
    val gpuLayers: Int = 0  // 0 for CPU only, > 0 for GPU acceleration
)

data class LLMState(
    val isLoading: Boolean = false,
    val isModelLoaded: Boolean = false,
    val error: String? = null,
    val messages: List<ChatMessage> = emptyList()
)
